base/dataset.go

```diff
@@ -265,3 +263,32 @@ func DatasetPodBodyFile(store cafs.Filestore, dsp *dataset.DatasetPod) (cafs.Fil
 
 	return cafs.NewMemfileReader(filepath.Base(dsp.BodyPath), file), nil
 }
+
+// ConvertBodyFormat rewrites a body from a source format to a destination format.
+func ConvertBodyFormat(bodyFile cafs.File, fromSt *dataset.Structure, toSt *dataset.Structure) (cafs.File, error) {
+	// Reader for entries of the source body.
+	r, err := dsio.NewEntryReader(fromSt, bodyFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Writes entries to a new body.
+	buffer := bytes.NewBufferString("")
+	w, err := dsio.NewEntryWriter(toSt, buffer)
+	if err != nil {
+		return nil, err
+	}
+
+	err = dsio.Copy(r, w)
+	if err != nil {
+		return nil, err
+	}
+	err = w.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the new format on the structure.
+	fromSt.Format = toSt.Format
+	return cafs.NewMemfileReader("", buffer), nil
```

Comment:
For the sake of consistency the name of MemFileReader should be `fmt.Sprintf("body.%s", toSt.Format)`.